File: src/main/java/io/vertx/ext/consul/v1/agent/Service.java
package io.vertx.ext.consul.v1.agent;

import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;

public interface Service {

  Service get(String serviceId, Handler<AsyncResult<ServiceInfo>> handler);

  Service healthByName(String name, Handler<AsyncResult<HealthStatus>> handler);

  Service healthById(String serviceId, Handler<AsyncResult<HealthStatus>> handler);

  Service register(RegisterPayload registerPayload, Handler<AsyncResult<Boolean>> handler);

  Service deregister(String serviceId, Handler<AsyncResult<Boolean>> handler);

  Service maintenance(String serviceId, Boolean enable, Handler<AsyncResult<Boolean>> handler);

  Service maintenance(String serviceId, Boolean enable, String reason, Handler<AsyncResult<Boolean>> handler);

}

File: vertx-consul-client/src/main/kotlin/io/vertx/kotlin/ext/consul/Service.kt
package io.vertx.kotlin.ext.consul

import io.vertx.ext.consul.Service

fun Service(
    address: String? = null,
  id: String? = null,
  name: String? = null,
  node: String? = null,
  nodeAddress: String? = null,
  port: Int? = null,
  tags: List<String>? = null): Service = io.vertx.ext.consul.Service().apply {

  if (address != null) {
    this.address = address
  }

  if (id != null) {
    this.id = id
  }

  if (name != null) {
    this.name = name
  }

  if (node != null) {
    this.node = node
  }

  if (nodeAddress != null) {
    this.nodeAddress = nodeAddress
  }

  if (port != null) {
    this.port = port
  }

  if (tags != null) {
    this.tags = tags
  }

}

